### PR TITLE
Fix invalid target framework error for Solution restore

### LINKED ISSUE #13772: "Sequence contains no matching element" when using a bad TF
### NuGet Product Used

MSBuild.exe

### Product Version

MSBuild version 17.9.8+b34f75857 for .NET Framework

### Worked before?

No

### Impact

It's more difficult to complete my work

### Repro Steps & Context

Error:
```
...\Current\bin\NuGet.targets(169,5): error : Sequence contains no matching element [<repo>\dirs.proj]
```

This happened when accidentally using:

```xml
    <TargetFramework>net8.0;net472</TargetFramework>
```

This is clearly a user error, but the error message could be more helpful. At the very least it should point to the problematic project instead of the root project for the restore.

I can only get this specific error to repro under certain conditions which I'm not able to fully understand. eg it manifests this way when restoring from the root. When restoring the project directly I helpfully get:

>NETSDK1046: The TargetFramework value 'net8.0;net472' is not valid. To multi-target, use the 'TargetFrameworks' property instead.

I would guess it maybe only manifests when a projects depends on the problematic project? Not super sure though.

Attaching a debugger shows that the exception is thrown here:
image

And the stack in case it's helpful:
image

I can provide more specific details (specific internal repo and commit) offline if desired.

### Verbose Logs

_No response_

### PR DESCRIPTION
**Pull request by martinrrm**
# Bug

Fixes: https://github.com/NuGet/Home/issues/13772

## Description
There is a difference in behavior when doing a restore in Solution vs Project with an invalid target framework in the `<TargetFramework>` tag.  When doing it in the project Level, sdk will run a validation before doing restore in https://github.com/dotnet/sdk/blob/5ab26633bdf4676efa25140a31a3d6182efb02f9/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets#L92.

The problem is that Solution level does not this validation (since it doesn't run a restore for each project) and will fail with 
```
...\Current\bin\NuGet.targets(169,5): error : Sequence contains no matching element [<repo>\dirs.proj]
```

This PR adds a validation and return the same error message as the project level. Currently, I decided to error with a NuGet error but we could also add `NETSDK1046` in the error message.

## PR Checklist

- [x] Meaningful title, helpful description and a linked NuGet/Home issue
- [x] Added tests
- [x] Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs [hunk 1]
```diff
@@ -454,6 +454,12 @@ private static IEnumerable<TargetFrameworkInformation> GetTargetFrameworkInforma
             foreach (var item in GetItemByType(items, "TargetFrameworkInformation"))
             {
                 var frameworkString = item.GetProperty("TargetFramework");
+                if (frameworkString.Contains(';'))
+                {
+                    var error = RestoreLogMessage.CreateError(NuGetLogCode.NU1001, string.Format(CultureInfo.CurrentCulture, Strings.Error_InvalidTargetFramework, frameworkString));
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs]: I know this is in draft, and we had chatted about validating the different scenarios. 
If we're going to add something like this, I think static graph might need some changes too.

Comment (martinrrm) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs]: Thanks for letting me know, I'll make sure I update static graph too

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs [hunk 2]
```diff
@@ -458,6 +458,12 @@ private static IEnumerable<TargetFrameworkInformation> GetTargetFrameworkInforma
             foreach (var item in GetItemByType(items, "TargetFrameworkInformation"))
             {
                 var frameworkString = item.GetProperty("TargetFramework");
+                if (frameworkString.Contains(';'))
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs:461]: Throwing from here probably leads to a bad failure in NuGet.exe?

In general, NuGet.exe expects a dg spec to create an this fails that. 

Can you double check NuGet.exe?

**Review by nkolev92 — COMMENTED** [review]
There's some test failures that seem like actual failures.
It's also worth calling out in the description that this is only a CLI problem in the PR description and that VS errors are way better.

File: test/NuGet.Core.Tests/NuGet.Build.Tasks.Console.Test/MSBuildStaticGraphRestoreTests.cs [hunk 3]
```diff
@@ -101,6 +101,19 @@ public void GetOriginalTargetFrameworks_WhenTargetFramworksSpecified_HasCorrectT
             actual.Should().BeEquivalentTo(expected);
         }
 
+        [Theory]
+        [InlineData("net40;net45")]
+        [InlineData("net40;net45 ; netstandard2.0 ")]
+        public void GetOriginalTargetFramework_WhenTargetFramworkSpecified_HasCorrectTargetFramework(string targetFrameworks)
```
Comment (jgonz120) [inline test/NuGet.Core.Tests/NuGet.Build.Tasks.Console.Test/MSBuildStaticGraphRestoreTests.cs:107]:
```suggestion
        public void GetOriginalTargetFramework_WhenMultipleTargetFramworkSpecified_ThrowRestoreCommandException(string targetFrameworks)
```

Might want to change the name of this test.

**Review by jgonz120 — APPROVED** [review]

**Pull request closed without merging**